src/Generator/Generator.php

```diff
@@ -74,11 +75,14 @@ public function generateMigration(
         string|null $up = null,
         string|null $down = null,
     ): string {
-        $mch = [];
-        if (preg_match('~(.*)\\\\([^\\\\]+)~', $fqcn, $mch) === 0) {
+        $mch         = [];
+        $matchResult = preg_match('~(.*)\\\\([^\\\\]+)~', $fqcn, $mch);
+        if ($matchResult === 0) {
```

Comment:
I've had the same problem in DBAL. If you change this check to…

```php
if ($matchResult !== 1) {
```

… you should not need the assert. Alternatively, we could think about using the composer/pcre library which has a saner API.